File: app/src/main/java/com/zeyad/generic/genericrecyclerview/adapter/screens/user/detail/Repository.java
package com.zeyad.generic.genericrecyclerview.adapter.screens.user.detail;


import com.zeyad.generic.genericrecyclerview.adapter.screens.user.list.User;

import org.parceler.Parcel;

/**
 * @author zeyad on 1/25/17.
 */
@Parcel
public class Repository {//extends RealmObject {
    //    @SerializedName("id")
    int id;
    //    @SerializedName("name")
    String name;
    //    @SerializedName("full_name")
    String fullName;
    //    @SerializedName("owner")
    User owner;

    public Repository() {
    }

    public static boolean isEmpty(Repository repository) {
        return repository == null || repository.name == null && repository.fullName == null &&
                repository.owner == null;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getFullName() {
        return fullName;
    }

    public void setFullName(String fullName) {
        this.fullName = fullName;
    }
}

File: app/src/main/java/com/zeyad/app/screens/detail/Repository.kt
package com.zeyad.app.screens.detail

import android.os.Parcelable
import com.google.gson.annotations.SerializedName
import com.zeyad.app.screens.User
import io.realm.RealmObject
import kotlinx.android.parcel.Parcelize

@Parcelize
open class Repository(@SerializedName("id")
                      var id: Int = 0,
                      @SerializedName("name")
                      var name: String = "",
                      @SerializedName("owner")
                      internal var owner: User? = null) : RealmObject(), Parcelable
